pkg/kubernetes/kubernetes.go

```diff
@@ -94,6 +90,39 @@ func GetKubeConfig(ctx context.Context) (*restclient.Config, error) {
 	return config, nil
 }
 
+// getKubeConfigPath returns the Kubeconfig path from the environment variable KUBECONFIG.
+// If the kubeconfig flag is supplied as an argument to the process, then it overrides the value
+// obtained from the KUBECONFIG environment variable and uses the supplied flag value.
+func getKubeConfigPath(ctx context.Context) string {
+	log := logger.GetLogger(ctx)
+	// Get the Kubeconfig path from the environment variable
+	kubecfgPath := os.Getenv(clientcmd.RecommendedConfigPathEnvVar)
+	log.Infof("Kubeconfig path obtained from environment variable %q: %q",
+		clientcmd.RecommendedConfigPathEnvVar, kubecfgPath)
```

Comment:
Can we move this code to line 114 instead?

Reply:
nit: +1. Since flag takes preference over env variable, we can check it in that order. We can probably set a `bool` variable to see if the value from flag is not empty, and then only check env variable if this value is not set to true.

Reply:
This will help debug issue and give clarity about how kubeconfig path is picked. We were having trouble identifying issue without these intermediate logs.

Reply:
done.

moved checking env after reading flag value.